app/src/chainlit.py

```diff
@@ -225,3 +243,29 @@ def _get_retrieval_metadata(result: OnMessageResult) -> dict:
         ],
         "raw_response": result.response,
     }
+
+
+async def _batch_proccessing(file: AskFileResponse) -> None:
+    await cl.Message(
+        author="backend",
+        content="Received file, processing...",
+    ).send()
+
+    try:
+        engine: chat_engine.ChatEngineInterface = cl.user_session.get("chat_engine")
+        result_file_path = await batch_process(file.path, engine)
```

Comment:
`batch_process()` returns the full path to the result file, including the filename, right? And it's being passed to `cl.File` on line 263 as the folder `path` -- interesting.

Reply:
Yup, then cl.File exposes the (local file path) on a public URL like e.g., `chat/project/file/d1c04d6c-3439-4c0f-820f-cac977cb0c30?session_id=ed644005-9fdd-440e-936d-dee1eb7f87cb`